pandas/core/generic.py

```diff
@@ -5172,9 +5196,23 @@ def __finalize__(self, other, method=None, **kwargs):
             types of propagation actions based on this
 
         """
+        from pandas.core.reshape.concat import _Concatenator
```

Comment:
These are just the minimal changes to `NDFrame.__finalize__` to get stuff working for my needs.

I'll be adding more `method`s later (`__array_ufunc__`, binary ops, etc.)

I'll remove these imports and the asserts.